### PR TITLE
[dagster-dbt][dagster-airbyte] Safer generation of output names

### PR DESCRIPTION
**Pull request by OwenKephart**
## Summary & Motivation

Output names must be valid python identifiers. It's possible to have dbt models or airbyte assets with hyphens in their names. If users had such models in the past, we would error when attempting to generate an asset. Now, we sanitize the identifiers.

Resolves: https://github.com/dagster-io/dagster/issues/15607

### CODE REVIEW

**Review by rexledesma — CHANGES_REQUESTED** [review]
Could we add a test?

**Comment by OwenKephart** [comment]
@rexledesma specifically for the dbt side of things? I'd prefer not to in this specific case, for a couple reasons:

1. There's no real chance of this introducing new bugs, as we were guaranteed to fail in this case in the past (so it can't get worse!). Any asset that does not have a hyphen in its name will be untouched by this update. We also do already do a very similar thing in input_name_fn (the thing above the modified function).
2. Modifying the existing test dbt project to include a model that has a hyphen name would require updating a bunch of tests, as we currently have a lot of assertions hardcoded into our tests about the precise keys that we're expecting to load. Similarly, adding a new one-off project for this purpose seems unnecessary.

I did however manually confirm that this works with a modified project (and that it failed without this change). Thoughts on if that's sufficient?

Definitely more amenable to adding a test to the airbyte side of things (as that's a larger change), although it might involve similar headaches to (2) (i'll defer to @benpankow on that one), and (1) also applies there.

File: python_modules/libraries/dagster-dbt/dagster_dbt/utils.py [hunk 1]
```diff
@@ -44,7 +44,7 @@ def input_name_fn(dbt_resource_props: Mapping[str, Any]) -> str:
 
 
 def output_name_fn(dbt_resource_props: Mapping[str, Any]) -> str:
-    return dbt_resource_props["unique_id"].split(".")[-1]
+    return dbt_resource_props["unique_id"].split(".")[-1].replace("-", "_")
```
Comment (rexledesma) [inline python_modules/libraries/dagster-dbt/dagster_dbt/utils.py:47]: Adding a comment on why we are doing this transformation would be nice

**Review by rexledesma — APPROVED** [review]
Ok. Deferring to @benpankow on the Airbyte changes.

File: python_modules/libraries/dagster-dbt/dagster_dbt/utils.py (hunk 1 above)
Comment (OwenKephart) [reply python_modules/libraries/dagster-dbt/dagster_dbt/utils.py:47]: sounds good

**Review by benpankow — APPROVED** [review]
this looks good to me, it would be nice if we could add a table name with a hyphen to the airbyte tests but not blocking

**Pull request merged**